Firmware/Marlin_main.cpp

```diff
@@ -1361,6 +1411,7 @@ void loop()
   manage_heater();
   isPrintPaused ? manage_inactivity(true) : manage_inactivity(false);
   checkHitEndstops();
+  host_keepalive();
```

Comment:
It seems that this is the only place where host_keepalive() function is called. Or am I missing something? This is inside void loop() function which is blocked when for example some blocking gcode is executed. host_keepalive() function should be called in function which takes care about managing heaters/steppers.